langchain4j/runtime/src/main/java/io/quarkiverse/flow/langchain4j/workflow/AbstractFlowAgentService.java

```diff
@@ -0,0 +1,134 @@
+package io.quarkiverse.flow.langchain4j.workflow;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import dev.langchain4j.agentic.internal.AbstractAgentInvocationHandler;
+import dev.langchain4j.agentic.internal.AbstractService;
+import dev.langchain4j.agentic.internal.AgentExecutor;
+import dev.langchain4j.agentic.internal.AgentSpecification;
+import dev.langchain4j.agentic.internal.AgenticScopeOwner;
+import dev.langchain4j.agentic.scope.AgenticScope;
+import dev.langchain4j.agentic.scope.DefaultAgenticScope;
+import io.quarkiverse.flow.internal.WorkflowNameUtils;
+import io.quarkiverse.flow.internal.WorkflowRegistry;
+import io.serverlessworkflow.api.types.Workflow;
+import io.serverlessworkflow.fluent.func.FuncDoTaskBuilder;
+import io.serverlessworkflow.fluent.func.FuncWorkflowBuilder;
+import io.serverlessworkflow.impl.WorkflowDefinition;
+import io.serverlessworkflow.impl.WorkflowDefinitionId;
+import io.serverlessworkflow.impl.WorkflowModel;
+
+/**
+ * Common Flow-backed implementation for LC4J workflow services
+ * (sequential, parallel, conditional, loop, ...).
+ * <p>
+ * Subclasses are responsible only for:
+ * - which LC4J service interface they implement (SequentialAgentService, ParallelAgentService, ...)
+ * - how they wire AgentExecutors into FuncWorkflowBuilder in {@link #doWorkflowTasks(List)}.
+ */
+public abstract class AbstractFlowAgentService<T, S> extends AbstractService<T, S> {
```

Comment:
Related to [this comment](https://github.com/quarkiverse/quarkus-flow/pull/107/files#r2601544231) unfortunately now I'm 100% sure that the introduction of the `Planner` abstraction in langchain4j 1.9.1 broke this integration.

Reply:
@mariofusco it will broke, but I'll change it on the upgrade. If I can't upgrade, I'll point that out to you and we can work together on a solution.

FWIW, the latest version of quarkus-langchain was based on 1.8.x.